src/helics/core/federate_id.hpp

```diff
@@ -15,6 +15,51 @@ namespace helics
 /** base date type for id types*/
 using identififier_base_type = int32_t;
 
+/** class defining a local_federate_id
```

Comment:
Is there a reason this was shifted in the file?

Reply:
This was moved so base_type and global_broker_id_shift would both be available for the globalIndex method added to local_federate_id (inverse of global_federate_id's localIndex). 
I simply moved all the constexpr together as a group, to avoid scattering them throughout the file.